internal/dao/container.go

```diff
@@ -46,14 +46,47 @@ func (c *Container) List(ctx context.Context, _ string) ([]runtime.Object, error
 	if err != nil {
 		return nil, err
 	}
-	res := make([]runtime.Object, 0, len(po.Spec.InitContainers)+len(po.Spec.Containers))
-	for _, co := range po.Spec.InitContainers {
-		res = append(res, makeContainerRes(co, po, cmx[co.Name], true))
+
+	type containerGroup struct {
```

Comment:
I don't think we need to introduce these groups. If we modified makeContainerRes signature to take in a kind (I,M,E) and an index should be sufficient. Then getContainerStatus can just look up statuses based on kind and index.
What do you think?

Reply:
@derailed It might be useful to add [sidecar containers](https://kubernetes.io/docs/concepts/workloads/pods/sidecar-containers/) (S) to the type list. This would help solve this issue #2912

Reply:
@derailed It could be done that way, but I introduced these groups to avoid having the logic spread across `List()` and `getContainerStatus()` with  `for` loops in each that both need to know about different container types. Since `getContainerStatus()` is only used in this one place, it makes it simpler to have the logic in one place and IMHO is easier to read and extend. I purposely made this new type private to this one method so it's not introducing a new global concept.

@ffais yes, I think that should be do-able. (This is actually a good example of how this is easier to extend by adding a predicate to the group to differentiate between sidecars and other init containers.)

Reply:
@derailed well, it looks like you already pulled in your own way of doing things in https://github.com/derailed/k9s/pull/2955/commits/b70b32c5a80cc96ab01d6f4d84559ec3a9142c36. I'll close this PR and the [issue](https://github.com/derailed/k9s/pull/2878).